src/platform/src/main/java/org/geoserver/platform/resource/Resources.java

```diff
@@ -593,6 +595,12 @@ public static Resource fromURL(Resource baseDirectory, String url) {
         if (url.startsWith("file:")) {
             url = url.substring(5); // remove 'file:' prefix
 
+            // revert encoded special characters and spaces
+            try {
+                url = URLDecoder.decode(url, "UTF-8");
+            } catch (UnsupportedEncodingException e) {
```

Comment:
Why is this exception being eaten? nom nom nom ...

Reply:
Q: This is detecting for relative paths, but misses the absolute path thing above?
Q: Does URLs.urlToFile need similar logic?

Reply:
The exception is eaten, because it cannot occur.
See https://docs.oracle.com/javase/8/docs/api/java/nio/charset/Charset.html
"Every implementation of the Java platform is required to support the following standard charsets. "

As explained on the jira ticket, absolute paths already support special characters, and only relative ones fail. This is exactly because URLS.urlTOFile already has the logic, and it is being called for absolute paths.